### DISCUSSION TITLE
Automatic include file resolved via require

### DISCUSSION THREAD

**Discussion by theKashey** [vercel/vercel#4003] [General] [answered]
### Current behaviour
https://zeit.co/docs/runtimes#advanced-usage/technical-details/including-additional-files
```js
const file = readFileSync(join(__dirname, 'config', 'ci.yml'), 'utf8');
// The use of __dirname is necessary to read a file relative to the current file.
```

### Better behaviour
use [require.resolve](https://nodejs.org/api/modules.html#modules_require_resolve_request_options)
```js
const file = readFileSync(require.resolve('./config/ci.yml'), 'utf8');
```

### What is not good about it
It already works at `now dev`, but not in prod.

**Comment by paulogdm** [comment]
Thanks for the suggestion!

cc @styfle

**Comment by guybedford** [comment]
`require.resolve` should be fine to use here and should be supported - if not that would a bug in node-file-trace.

**Reply by styfle** [reply]
Confirmed, this is working as expected

Source: https://zeit.co/testing-internal/trace-resolve/i0jzw6kxf/source
Demo: https://trace-resolve-i0jzw6kxf.now.sh/api

**Comment by theKashey** [comment, marked as answer]
Yep. It was a mistake of mine - I was trying to read something from `public` directory, which does not exist for functions.
See #4010
